test/unit/model/helpers/downloader-test.js

```diff
@@ -271,4 +272,31 @@ describe('Downloader', function() {
       expect(successHandler).to.be.calledOnce;
     });
   });
+
+  describe('restartDownload', function(){
+    let options = 'http://example.com/jdk.zip';
+
+    it('should change downloader status from \'Download Failed\' to \'Downloading\'', function(){
+      expect(downloader.restartDownload()).to.be.equal(undefined);
+    });
+
+    it('should create new WriteStream instance for failed download entries', function(){
+
+      let spy = sandbox.spy(fs, 'createReadStream');
+      downloader.restartDownload();
+      expect(spy).not.called;
+    });
+
+    it('should call authDownload for entries that requires authentication', function(){
+      let donwloadauth = downloader.downloadAuth(options, 'username', 'password', 'key', 'sha');
```

Comment:
Every test inside `describe('restartDownload')` is supposed to have similar structure. 
Each test would have:
* initialization part where you prepare for testing specific invariant/scenario
* `downloader.restartDownload()` call 
* verification part where it would check all expectations

This test and next one should test restartDownload method and verify that it behaves as expected for different kinds of downloaded resources. We have two kinds of resources here one that requires authentication and one that does not.

you can use the same [lines 205 - 215](https://github.com/redhat-developer-tooling/developer-platform-install/pull/476/files#diff-762967bfd2ff9393807ffee1c812a190R205) code as example for initialization part. Bot not just copy it as is :), add required changes to control restartDownload() behavior.